common/djangoapps/third_party_auth/saml.py

```diff
@@ -112,10 +114,20 @@ def get_user_details(self, attributes):
         """
         details = super(EdXSAMLIdentityProvider, self).get_user_details(attributes)
         extra_field_definitions = self.conf.get('extra_field_definitions', [])
+        
         details.update({
             field['name']: attributes[field['urn']][0] if field['urn'] in attributes else None
             for field in extra_field_definitions
         })
+        
+        # TODO: ADD FEATURE FLAG
+        username_generator_settings = self.conf.get('USERNAME_GENERATOR', {})
+
+        fullname = details['fullname']
+        username_generator = UsernameGenerator(username_generator_settings)
+        username = username_generator.hint_username(fullname)
+        details.update({'username': username})
```

Comment:
If `details['username']` already exists, should we skip the generator completely and just use the original value?